### PR TITLE
Update to use DependencyDescriptor

### PR DESCRIPTION
**Pull request by murphm8**
Depends on colcon/colcon-core#106

### CODE REVIEW

**Comment by murphm8** [comment]
I tested this by running `colcon list` and `colcon list --topological-graph` on a workspace. I compared it to the output of a container with colcon installed from PyPi.

File: colcon_package_information/verb/list.py [hunk 1]
```diff
@@ -120,10 +120,12 @@ def main(self, *, context):  # noqa: D102
                     if j == i:
                         # package i is being processed
                         lines[j] += '+'
-                    elif shown_decorators[j].descriptor.name in depends[i]:
+                    elif shown_decorators[j].descriptor.name \
+                            in set(map(lambda d: d.name, depends[i])):
```
Comment (dirk-thomas) [inline colcon_package_information/verb/list.py]: Can you rewrite this condition to:
```
elif (
    shown_decorators[j].descriptor.name in
    {d.name for d in depends[i]}
):
```

(avoiding the double indentation as well as the nested set/map/lamda).

Same below.

**Comment by murphm8** [comment]
Updated to use dependency_name() for backwards compatibility. There will need to be a version bump of the colcon-core dependency in setup.cfg.

**Comment by dirk-thomas** [comment]
This doesn't look backward compatible to me. When a user installs the upcoming version of `colcon-core` the existing release of this package would fail to work, no? So the user would also need this patch or am I understanding this patch wrong?

**Comment by murphm8** [comment]
Updating colcon-core itself would not break this package. If the customer updated colcon-core and colcon-ros to my changes, then this package would break since colcon-ros would start using DependencyDescriptors. Based on my limited knowledge of Python, I don't know a way to prevent this type of breakage.

**Comment by murphm8** [comment]
This is no longer necessary since DependencyDescriptor is a str.

**Pull request closed without merging**